integrations/aws-v3/aws/core/exporters/s3/builder.py

```diff
@@ -0,0 +1,15 @@
+from aws.core.exporters.s3.models import S3Bucket, S3BucketProperties
+from typing import Dict, Any, Self
+
+
+class S3BucketBuilder:
+    def __init__(self, name: str) -> None:
+        self._bucket = S3Bucket(Identifier=name, Properties=S3BucketProperties())
+
+    def with_data(self, data: Dict[str, Any]) -> Self:
+        for k, v in data.items():
+            setattr(self._bucket.Properties, k, v)
+        return self
+
+    def build(self) -> S3Bucket:
+        return self._bucket
```

Comment:
feels kinda general, might be a good idea to abstract it so we won't have to repeat it on every resource